android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java

```diff
@@ -240,7 +249,11 @@ public boolean setKeyboard(String packageID, String keyboardID, String languageI
     boolean retVal = true;
     String keyboardVersion = KMManager.getLatestKeyboardFileVersion(getContext(), packageID, keyboardID);
     if (!KMManager.shouldAllowSetKeyboard() || keyboardVersion == null) {
-      Toast.makeText(context, "Invalid keyboard! Loading default", Toast.LENGTH_LONG).show();
+      if(keyboardVersion == null) {
+        Toast.makeText(context, "Invalid keyboard - no version specified! Loading default", Toast.LENGTH_LONG).show();
+      } else {
+        Toast.makeText(context, "Invalid keyboard! Loading default", Toast.LENGTH_LONG).show();
```

Comment:
How can this happen -- can it happen due to user error? Also, I'd like the toast to give the packageID, keyboardID, languageID so the error message is more useful to us/user.